### ISSUE TITLE
Starting Network in Python file is different that starting it via command line

### ISSUE THREAD

**Issue by stephendiadamo** [SoftwareQuTech/SimulaQron#214]
If I start the network using this method:

```
from simulaqron.network import Network as SimulaNetwork      
nodes = ["Alice", "Bob"]
sim_network = SimulaNetwork(nodes=nodes, force=True)
sim_network.start()
```

Then I can only send qubits in the direction ```Alice -> Bob```. I cannot send qubits in the direction  ```Bob -> Alice```. Alternatively if I run 

```
from simulaqron.network import Network as SimulaNetwork      
nodes = ["Bob", "Alice"]
sim_network = SimulaNetwork(nodes=nodes, force=True)
sim_network.start()
```

I can only send qubits in the direction  ```Bob -> Alice``` and not the other way.
The error is:
```
  File ".../lib/python3.7/site-packages/cqc/pythonLib.py", line 1068, in recvQubit
    message = self.readMessage()
  File ".../lib/python3.7/site-packages/cqc/pythonLib.py", line 829, in readMessage
    data = self._s.recv(maxsize)
OSError: [Errno 9] Bad file descriptor
```

If instead I start  simulaqronfrom command line, this problem doesn't exist.  That is, if I run:
``` simulaqron start --nodes Alice,Bob --force ```.

**Comment by AckslD** [comment]
Hi @stephendiadamo ! Thanks for pointing out this bug, I'm not directly sure why this is happening but would have to investigate. What's your version of `simulaqron` and `cqc`?

**Comment by stephendiadamo** [comment]
Thanks for the help. 
The versions are:
```
cqc==3.0.2
simulaqron==3.0.4
```

**Comment by AckslD** [comment]
@stephendiadamo Could you try to guard you code with a `if __name__ == '__main__':` check and see if it makes a difference? See also #220

**Comment by stephendiadamo** [comment]
I'm running the code in a test suite using nose2, so I'm not running the code in a main file. The error in #220 I don't experience, the network does start. For me it's that it's not possible to send qubits from Bob -> Alice depending on if I initialize Alice or Bob first. I see in #220 though that the topology is defined., that's something I didn't do. Maybe if I add that it'll fix the problem. I'll try it and give an update.

**Comment by stephendiadamo** [comment]
Ah.. Setting the topology didn't fix it.

**Comment by AckslD** [comment]
@stephendiadamo I am unable to reproduce this issue. Could you share your full code? If you don't want to do it here you can also email me.

**Comment by AckslD** [comment]
Hi @stephendiadamo ! I finally managed to find what was going on here. The issue was that when we start a network, we check to see if it's up and running by connecting to it using the `CQCConnection`, this then increased the used application IDs (this was a bug) and since the `CQCConnections` you then open in your example are in the same file and share this class attribute they got assigned a application ID of 3-7 or so. This is why `Alice` could not recieve the qubit sent from Alice, since Alice sent it to an application on node `Bob` with application ID 0 but `Bob` had something higher. You can try that this is the case by adding for example
```python
Bob.sendQubit(q_send, 'Alice', remote_appID=Alice._appID)
```
However, this is probably not what you want but just to check that it works. Anyway the bug is now fixed in `cqc` version 3.0.4 which is now required by `simulaqron` version 3.0.8.

Let me know if it still does not work for you.

**Comment by stephendiadamo** [comment]
Great thanks! The update fixed it.
I'll close the issue now.

Thanks again.
